### ISSUE TITLE
Can't build recent version on macOS or Ubuntu

### ISSUE THREAD

**Issue by richb-hanover** [ExceLint/ExceLint-addin#35]
**Describe the bug**
I get a couple error messages when attempting to build and run locally the ExceLint binary.

* I'm using a MacBook Pro, macOS 10.15.7
* Node 16.14.0
* I get "node: --no-experimental-fetch is not allowed in NODE_OPTIONS" when I enter `npm install`
* If I set NODE_OPTIONS="", `npm install` gives an error "Cannot read properties of null (reading 'resolve')"

**To Reproduce**
1. Follow the steps in the README; git clone, cd <directory>, etc.
2. The full terminal session is shown below

**Expected behavior**
I expected not to see error messages

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS 10.15.7
 - Browser -none-
 - Version -none-

**Smartphone (please complete the following information):**
 - Browser [e.g. stock browser, Safari]

**Additional context**
```
√ ExceLint-addin % nvm list
       v12.18.4
        v14.8.0
       v14.17.0
       v16.13.1
->     v16.14.0
         system
default -> 14.8.0 (-> v14.8.0)
iojs -> N/A (default)
unstable -> N/A (default)
node -> stable (-> v16.14.0) (default)
stable -> 16.14 (-> v16.14.0) (default)
lts/* -> lts/gallium (-> v16.14.0)
lts/argon -> v4.9.1 (-> N/A)
lts/boron -> v6.17.1 (-> N/A)
lts/carbon -> v8.17.0 (-> N/A)
lts/dubnium -> v10.24.1 (-> N/A)
lts/erbium -> v12.22.10 (-> N/A)
lts/fermium -> v14.19.0 (-> N/A)
lts/gallium -> v16.14.0
√ ExceLint-addin % export NODE_OPTIONS=--no-experimental-fetch
√ ExceLint-addin % npm install
node: --no-experimental-fetch is not allowed in NODE_OPTIONS
?9 ExceLint-addin % export NODE_OPTIONS=""
√ ExceLint-addin % npm install
npm ERR! Cannot read properties of null (reading 'resolve')

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/richb/.npm/_logs/2022-06-12T20_54_37_578Z-debug-0.log
?1 ExceLint-addin %
```

**Comment by richb-hanover** [comment]
I had trouble on Ubuntu 22.04 as well. I'm running a fairly clean Parallels VM on my MacBook of Ubuntu 22.04. I installed git, node, and then followed the instructions on the README. `npm install` worked fine (but there were > 75 audit warnings). This is what i saw when I used `npm run start-local`:

What other troubleshooting information could I provide? Many thanks!

```
richb@richb-Parallels-Virtual-Platform:~/Documents/github/ExceLint-addin$ npm run start-local

> exce-lint@1.0.0 start-local
> webpack serve --inline --host localhost --port 3000 --config config/webpack.dev-local.js --progress


  Replace Autoprefixer browsers option to Browserslist config.
  Use browserslist key in package.json or .browserslistrc file.

  Using browsers option can cause errors. Browserslist config can
  be used for Babel, Autoprefixer, postcss-normalize and other tools.

  If you really need to use option, rename it to overrideBrowserslist.

  Learn more at:
  https://github.com/browserslist/browserslist#readme
  https://twitter.com/browserslist


10% building 5/5 modules 0 activeℹ ｢wds｣: Project is running at https://localhost:3000/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from /home/richb/Documents/github/ExceLint-addin/dist
ℹ ｢wds｣: 404s will fallback to /index.html
✖ ｢wdm｣: Hash: 8e2ed6884bef2f395190
Version: webpack 4.46.0
Time: 22336ms
Built at: 06/12/2022 6:16:16 PM
                                Asset      Size  Chunks               Chunk Names
          app.8e2ed6884bef2f395190.js  14.3 MiB       0  [immutable]  app
function-file.8e2ed6884bef2f395190.js   921 KiB       1  [immutable]  function-file
       vendor.8e2ed6884bef2f395190.js    10 MiB       2  [immutable]  vendor
Entrypoint vendor = vendor.8e2ed6884bef2f395190.js
Entrypoint app = app.8e2ed6884bef2f395190.js
Entrypoint function-file = function-file.8e2ed6884bef2f395190.js
[1] multi ../node_modules/webpack-dev-server/client?https://localhost:3000 ../node_modules/webpack/hot/dev-server.js react react-dom core-js office-ui-fabric-react 88 bytes {2} [built]
[2] multi ../node_modules/webpack-dev-server/client?https://localhost:3000 ../node_modules/webpack/hot/dev-server.js react-hot-loader/patch ./index.tsx 64 bytes {0} [built]
[5] multi ../node_modules/webpack-dev-server/client?https://localhost:3000 ../node_modules/webpack/hot/dev-server.js ../function-file/function-file.ts 52 bytes {1} [built]
[../function-file/function-file.ts] 1.32 KiB {1} [built]
[../node_modules/core-js/es/index.js] 8.89 KiB {2} [built]
[../node_modules/core-js/index.js] 123 bytes {2} [built]
[../node_modules/core-js/internals/path.js] 71 bytes {2} [built]
[../node_modules/core-js/proposals/index.js] 21 bytes {2} [built]
[../node_modules/office-ui-fabric-react/lib/index.js] ../node_modules/office-ui-fabric-react/lib/index.js + 625 modules 2.17 MiB {0} {2} [built]
    | ../node_modules/office-ui-fabric-react/lib/index.js 2.6 KiB [built]
    | ../node_modules/office-ui-fabric-react/lib/ActivityItem.js 89 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Autofill.js 81 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Announced.js 83 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Breadcrumb.js 85 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Button.js 77 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/ButtonGrid.js 84 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Calendar.js 81 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Callout.js 79 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Check.js 75 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Checkbox.js 81 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/ChoiceGroup.js 87 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Coachmark.js 83 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/Color.js 74 bytes [built]
    | ../node_modules/office-ui-fabric-react/lib/ColorPicker.js 87 bytes [built]
    |     + 611 hidden modules
[../node_modules/react-dom/index.js] 1.33 KiB {0} {2} [built]
[../node_modules/react-hot-loader/patch.js] 229 bytes {0} [built]
[../node_modules/react/index.js] 190 bytes {0} {2} [built]
[../node_modules/webpack-dev-server/client/index.js?https://localhost:3000] ../node_modules/webpack-dev-server/client?https://localhost:3000 4.29 KiB {0} {1} {2} [built]
[../node_modules/webpack/hot/dev-server.js] 1.59 KiB {0} {1} {2} [built]
[./index.tsx] 2.04 KiB {0} [built]
    + 766 hidden modules

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/

ERROR in ./components/App.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/colorize' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/App.tsx 7:19-58
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/App.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/config' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/App.tsx 11:17-54
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/Content.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/config' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/Content.tsx 7:17-54
 @ ./components/App.tsx
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/App.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/exceljson' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/App.tsx 9:20-60
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/App.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/excelutils' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/App.tsx 8:21-62
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/Content.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/excelutils' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/Content.tsx 6:21-62
 @ ./components/App.tsx
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/App.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/option' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/App.tsx 13:17-54
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in ./components/App.tsx
Module not found: Error: Can't resolve './ExceLint-core/src/timer' in '/home/richb/Documents/github/ExceLint-addin/src/components'
 @ ./components/App.tsx 10:16-52
 @ ./index.tsx
 @ multi react-hot-loader/patch ./index.tsx

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(5,26)
      TS2307: Cannot find module './ExceLint-core/src/colorize' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(6,28)
      TS2307: Cannot find module './ExceLint-core/src/excelutils' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(7,27)
      TS2307: Cannot find module './ExceLint-core/src/exceljson' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(8,23)
      TS2307: Cannot find module './ExceLint-core/src/timer' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(9,24)
      TS2307: Cannot find module './ExceLint-core/src/config' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(10,23)
      TS2307: Cannot find module './ExceLint-core/src/ExceLintTypes' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(12,32)
      TS2307: Cannot find module './ExceLint-core/src/ExceLintTypes' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx
./components/App.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/App.tsx(13,36)
      TS2307: Cannot find module './ExceLint-core/src/option' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/Content.tsx
./components/Content.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/Content.tsx(3,28)
      TS2307: Cannot find module './ExceLint-core/src/excelutils' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/Content.tsx
./components/Content.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/Content.tsx(4,45)
      TS2307: Cannot find module './ExceLint-core/src/ExceLintTypes' or its corresponding type declarations.

ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/Content.tsx
./components/Content.tsx
[tsl] ERROR in /home/richb/Documents/github/ExceLint-addin/src/components/Content.tsx(5,24)
      TS2307: Cannot find module './ExceLint-core/src/config' or its corresponding type declarations.
Child HtmlWebpackCompiler:
                          Asset      Size  Chunks  Chunk Names
    __child-HtmlWebpackPlugin_0  7.26 KiB       0  HtmlWebpackPlugin_0
    __child-HtmlWebpackPlugin_1  7.22 KiB       1  HtmlWebpackPlugin_1
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_1 = __child-HtmlWebpackPlugin_1
    [../node_modules/html-webpack-plugin/lib/loader.js!../function-file/function-file.html] 1.14 KiB {1} [built]
    [../node_modules/html-webpack-plugin/lib/loader.js!./index.html] 1.19 KiB {0} [built]
ℹ ｢wdm｣: Failed to compile.

```

**Comment by dbarowy** [comment]
Hi @richb-hanover, thanks for the bug report.  I'm at a conference this week, but I'll try to reproduce this when I have a moment.

**Comment by richb-hanover** [comment]
And I'm happy to provide more troubleshooting info if needed. Thanks!

**Comment by richb-hanover** [comment]
Solved with #46 (Run a local copy with Docker) or #48 (use the version served by a Github Page)
